lint/location.go

```diff
@@ -14,43 +14,90 @@
 
 package lint
 
-import "fmt"
+import (
+	dpb "github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/jhump/protoreflect/desc"
+)
 
-// Location describes a location in a source code file.
+// PathLocation returns the precise location within a file for a given path.
+func PathLocation(f *desc.FileDescriptor, path []int32) *dpb.SourceCodeInfo_Location {
+	return sourceInfoRegistry.sourceInfo(f).findLocation(path)
+}
+
+// SyntaxLocation returns the location of the syntax definition in a file
+// descriptor.
 //
-// Note: positions are one-based.
-type Location struct {
-	Start Position `json:"start_position" yaml:"start_position"`
-	End   Position `json:"end_position" yaml:"end_position"`
+// If the location can not be found (for example, because there is no syntax
+// statement), it returns beginning of file.
+func SyntaxLocation(f *desc.FileDescriptor) *dpb.SourceCodeInfo_Location {
+	return PathLocation(f, []int32{12}) // syntax == 12
 }
 
-// IsValid checks if the location is constructed properly and
-// returns true if so.
-func (l Location) IsValid() bool {
-	return l.Start.IsValid() &&
-		l.End.IsValid() &&
-		(l.End.Line > l.Start.Line ||
-			l.End.Line == l.Start.Line && l.End.Column >= l.Start.Column)
+// PackageLocation returns the location of the package definition in a file
+// descriptor.
+//
+// If the location can not be found (for example, because there is no package
+// statement), it returns beginning of file.
```

Comment:
Update this comment. I think it now returns `nil` if not found.

Reply:
Fixed.